### ISSUE TITLE
Streaming Service: enhanced support for external DB

### ISSUE THREAD

**Issue by alexsasharegan** [tootsuite/mastodon#11445]
tl;dr;

- Support `sslmode=require` in pg connection uri
- Support ssl config without requiring use of `DATABASE_URL`

### Pitch

I just finished setting up my instance on digitalocean (DO) using the managed postgres cluster and their spaces object storage. It appears DO supplies you with a convenient connection uri copy that uses the `sslmode=require` form rather than `ssl=true`. This appears to be convention going forward given that it appears in [PG docs v10 & v11](https://www.postgresql.org/docs/11/libpq-connect.html#id-1.7.3.8.3.6)

> For improved compatibility with JDBC connection URIs, instances of parameter ssl=true are translated into sslmode=require.

The code currently ignores `sslmode=require`, which leads to some confusing debugging to the less initiated. Perhaps this code should move to using this standard while maintaining compat with `ssl=true`: https://github.com/tootsuite/mastodon/blob/master/streaming/index.js#L48

Also, there may be a need for another env param to enable ssl mode. I have to use the `DATABASE_URL` to successfully inject the ssl conf, and this is not part of the canonical install instructions path.

### Motivation

I think the instructions are pretty awesome, but there's some room for improvement when incorporating managed services for DB and file hosting (also had a little spelunking to do for that). The snags are minor in the code, but cost me several hours.

**Comment by radiodario** [comment]
Hey @alexsasharegan [i just modified the code for the streaming server myself on my fork of mastodon to enable SSL](https://github.com/radiodario/100club/commit/47ce37ff853d7f0f1bac0d6ee31a9a5c07f31e86#diff-f9eb75341ce6bbfb6d0468e132684dc0b0d86cb8e26d9460602563cc46115c49) and use the native client rather than the node.js javascript one - i'm not sure what the most maintainable way of doing this is - since i just slightly hard coded it, but it took me a while to figure out that the streaming service uses a slightly different approach. 

We might want to re-open this and consider it for newer versions?

**Comment by rasca** [comment]
@dungsil it works in my end. We are using v3.5.5 . Where are you seeing that error? My clients are correctly subscribing to the websocket.

**Comment by k3rn3l3rr0r** [comment]
Thanks for this answer @rasca ... I was able to get my streaming to work using the following changes in the official docker-compose.yml shipped with v4.0.2...

```
    #command: node ./streaming    
    command:   
      - bash   
      - -c   
      - |  
        sed -i 's/.ssl = true/.ssl = { rejectUnauthorized: false }/g' ./streaming/index.js  
        node ./streaming  
```

and in .env.production just add `DB_SSLMODE=require`

This is nice, because i don't need to re-build my own image all the time, because of this simple change.
Don't know abut the security aspect though. As rejectUnauthorized should be set to true for SSL connections according to the docs.

**Comment by ElanHasson** [comment]
strange this worked for me when running on a droplet, but was not required on App Platform at DO.
